ompi/include/ompi/memchecker.h

```diff
@@ -128,7 +128,18 @@ static inline int memchecker_call (int (*f)(void *, size_t), const void * addr,
 
             while( pElem->elem.common.flags & OPAL_DATATYPE_FLAG_DATA ) {
                 /* now here we have a basic datatype */
-                f( (void *)(source_base + pElem->elem.disp), pElem->elem.count*pElem->elem.extent );
+                size_t blength = opal_datatype_basicDatatypes[pElem->elem.common.type]->size;
+                if ((size_t)pElem->elem.extent == blength) {
+                    /* block is made of contiguous basic datatype */
+                    f( (void *)(source_base + pElem->elem.disp), pElem->elem.count*pElem->elem.extent );
```

Comment:
It will be more readable if you use blength (even after the check 2 lines above).